### PR TITLE
[ePD] Fix action endpoints

### PR DESCRIPTION
**Pull request by reinbach**
[ch22437]
[ch22345]
[ch22747]

No longer allow POST requests
Add review/cancel/signature action endpoints

### CODE REVIEW

**Review by roman-karpovich — APPROVED** [review]

File: src/etools/applications/partners/views/interventions_v3_actions.py [hunk 1]
```diff
@@ -55,27 +58,174 @@ def update(self, request, *args, **kwargs):
...
+
+class PMPInterventionTerminateView(PMPInterventionMixin, InterventionDetailAPIView):
+    def update(self, request, *args, **kwargs):
+        if self.is_partner_staff():
+            return HttpResponseForbidden()
+        pd = self.get_object()
+        if pd.status == Intervention.TERMINATED:
+            raise ValidationError("PD has already been terminated.")
+        request.data.clear()
+        request.data.update({"status": Intervention.TERMINATED})
```
Comment (robertavram) [inline src/etools/applications/partners/views/interventions_v3_actions.py:167]: pending updates

**Review by robertavram — CHANGES_REQUESTED** [review]

File: src/etools/applications/partners/serializers/interventions_v3.py [hunk 2]
```diff
@@ -197,13 +197,18 @@ def get_available_actions(self, obj):
...
+                    available_actions.append("terminate")
             return available_actions
 
         # PD is assigned to UNICEF
         if obj.unicef_court:
             # UNICEF User with Senior Management Team
             if self._is_management():
-                available_actions.append("cancel")
+                if obj.status in [obj.DRAFT, obj.REVIEW, obj.SIGNATURE]:
+                    available_actions.append("cancel")
```
[the remark this answers is not in the document]
Comment (reinbach) [reply src/etools/applications/partners/serializers/interventions_v3.py]: :+1:

File: src/etools/applications/partners/serializers/interventions_v3.py [hunk 3]
```diff
@@ -197,13 +197,18 @@ def get_available_actions(self, obj):
 
         # available actions only provided in Development status
         if obj.status != obj.DRAFT:
+            # need to add terminate action for Senior Management team
```
[the remark this answers is not in the document]
Comment (reinbach) [reply src/etools/applications/partners/serializers/interventions_v3.py]: :+1:

File: src/etools/applications/partners/models.py [hunk 4]
```diff
@@ -2226,6 +2228,13 @@ def transition_to_active(self):
     def transition_to_signed(self):
         pass
 
+    @transition(field=status,
+                source=[DRAFT, REVIEW, SIGNATURE],
+                target=[CANCELLED],
+                conditions=[intervention_validation.transition_to_cancelled])
+    def transition_to_cancelled(self):
+        pass
```
[the remark this answers is not in the document]
Comment (reinbach) [reply src/etools/applications/partners/models.py:2236]: :+1:

File: src/etools/applications/partners/views/interventions_v3_actions.py [hunk 5]
```diff
@@ -78,6 +78,105 @@ def update(self, request, *args, **kwargs):
...
+                "pmp_v3:intervention-detail",
+                args=[pd.pk]
+            ),
+        }
+        send_notification_with_template(
+            recipients=recipients,
+            template_name='partners/intervention/unicef_reviewed',
+            context=context
+        )
+
```
[the remark this answers is not in the document]
Comment (reinbach) [reply src/etools/applications/partners/views/interventions_v3_actions.py]: :+1:

File: src/etools/applications/partners/serializers/interventions_v3.py [hunk 6]
```diff
@@ -197,13 +197,18 @@ def get_available_actions(self, obj):
...
+            if self._is_management():
```
[the remark this answers is not in the document]
Comment (reinbach) [reply src/etools/applications/partners/serializers/interventions_v3.py]: :+1:

File: src/etools/applications/partners/views/interventions_v3_actions.py (hunk 1 above)
Comment (reinbach) [reply src/etools/applications/partners/views/interventions_v3_actions.py]: :+1:

**Review by roman-karpovich — APPROVED** [review]

**Pull request merged**